tools/topology/m4/pcm.m4

```diff
@@ -107,6 +107,38 @@ define(`PCM_CAPTURE_ADD',
 `}', `fatal_error(`Invalid parameters ($#) to PCM_CAPTURE_ADD')')'
 )
 
+dnl PCM_CAPTURE_LP_ADD(name, pcm_id, capture)
+define(`PCM_CAPTURE_LP_ADD',
+`ifelse(`$#', `3',
+`SectionVendorTuples."$1_tuples_w" {'
+`       tokens "sof_stream_tokens"'
+`       tuples."bool" {'
+`               SOF_TKN_STREAM_CAPTURE_COMPATIBLE_D0I3        "true"'
+`       }'
+`}'
+`SectionData."$1_data_w" {'
+`       tuples "$1_tuples_w"'
+`}'
+`SectionPCM.STR($1) {'
+`'
+`	# used for binding to the PCM'
+`	id STR($2)'
+`'
+`	dai.STR($1 $2) {'
+`		id STR($2)'
+`	}'
+`'
+`	pcm."capture" {'
+`'
+`		capabilities STR($3)'
+`	}'
+`'
+`       data ['
+`               "$1_data_w"'
+`       ]'
+`}', `fatal_error(`Invalid parameters ($#) to PCM_CAPTURE_LP_ADD')')'
```

Comment:
can't we use the regular capture macro here? all you are doing is add support for one token.

Reply:
@plbossart do you mean add a param to
--PCM_CAPTURE_ADD(name, pcm_id, capture)
++PCM_CAPTURE_ADD(name, pcm_id, capture, d0i3_compatible)?

Reply:
I chose to add a new macro to avoid huge changes in other .m4, as most of others won't use this flag.

Reply:
I meant adding a macro that extends a previous one instead of redefining everything inside.
e.g 

PCM_CAPTURE_LP_ADD(name, pcm_id, capture)
  PCM_CAPTURE_ADD(name, pcm_id, capture)
  do0ix-specific parts

Reply:
@plbossart I think this is not easy to implement with m4 syntax, especially those d0ix-specific parts are in the middle of PCM_CAPTURE_ADD macro.

Reply:
@keyonjie doing this is easy enough, you need to have a internal macro that generates the alsaconf and two external public macros used for each type.

Reply:
OK, @lgirdwood @plbossart now figure it out.